org._3fz.mtoc.yml

```diff
@@ -0,0 +1,99 @@
+app-id: org._3fz.mtoc
+runtime: org.kde.Platform
+runtime-version: '6.9'
+sdk: org.kde.Sdk
+command: mtoc_app
+finish-args:
+    # Display
+    - --socket=wayland
+    - --socket=fallback-x11
+    - --share=ipc
+    - --device=dri
+
+    # GPU acceleration support
+    - --device=all
+    - --share=network
+
+    # Environment variables for Qt rendering
+    - --env=QT_QPA_PLATFORMTHEME=kde
+
+    # AMD GPU optimization
+    - --env=AMD_VULKAN_ICD=RADV
+    - --env=MESA_LOADER_DRIVER_OVERRIDE=radeonsi
+
+    # Audio
+    - --socket=pulseaudio
+
+    # File access - Music directory only
+    - --filesystem=xdg-music:rw
+
+    # D-Bus access for MPRIS
+    - --talk-name=org.mpris.MediaPlayer2.mtoc
+    - --own-name=org.mpris.MediaPlayer2.mtoc
+
+
+cleanup:
+  - /include
+  - /lib/pkgconfig
+  - /share/pkgconfig
+  - '*.la'
+  - '*.a'
+
+modules:
+  - name: utf8cpp
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DUTF8_TESTS=OFF
+      - -DUTF8_SAMPLES=OFF
+    sources:
+      - type: archive
+        url: https://github.com/nemtrif/utfcpp/archive/refs/tags/v4.0.5.tar.gz
+        sha256: ffc668a310e77607d393f3c18b32715f223da1eac4c4d6e0579a11df8e6b59cf
+    cleanup:
+      - '*'
+
+  - name: taglib
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DBUILD_SHARED_LIBS=ON
+      - -DBUILD_TESTING=OFF
+      - -DBUILD_EXAMPLES=OFF
+    sources:
+      - type: archive
+        url: https://github.com/taglib/taglib/releases/download/v2.0.2/taglib-2.0.2.tar.gz
+        sha256: 0de288d7fe34ba133199fd8512f19cc1100196826eafcb67a33b224ec3a59737
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+      - '*.la'
+
+  # Add mesa-utils for debugging
+  - name: mtoc
+    buildsystem: cmake
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+    sources:
+      - type: git
+        url: https://github.com/asa-degroff/mtoc.git
+        tag: mtoc-2.0.1-1
+        commit: ecd4df08b59ade282a537ac42c6c3075c1f39a56
+      - type: file
+        path: org._3fz.mtoc.metainfo.xml
+    post-install:
+      # Remove the original desktop file and icons installed by CMake
+      - rm -f /app/share/applications/mtoc.desktop
+      - rm -f /app/share/icons/hicolor/*/apps/mtoc.png
+      - rm -f /app/share/icons/hicolor/scalable/apps/mtoc.svg
+      - rm -f /app/share/pixmaps/mtoc.png
+      # Copy and modify desktop file to use correct icon name
+      - cp mtoc.desktop org._3fz.mtoc.desktop
+      - sed -i 's/Icon=mtoc/Icon=org._3fz.mtoc/' org._3fz.mtoc.desktop
```

Comment:
also never use sed to edit desktop files. The `desktop-file-edit` commands exist. Albeit with the comment above it's not even necessary.